catalog/core/catalog-core-api/src/main/java/ddf/catalog/source/SourceDescriptor.java

```diff
@@ -49,4 +51,11 @@
    * @return the last availability date
    */
   public Date getLastAvailabilityDate();
+
+  /**
+   * Get the list of actions associated with the {@link Source}.
+   *
+   * @return list of actions
+   */
+  List<Action> getActions();
```

Comment:
❗️ The source interface should not change.  Instead you need to create an `ActionRegistry` and `ActionProvider` where `T` is Source.
https://github.com/codice/ddf/blob/master/platform/action/core/platform-action-core-api/src/main/java/ddf/action/ActionRegistry.java
https://github.com/codice/ddf/blob/master/platform/action/core/platform-action-core-api/src/main/java/ddf/action/ActionProvider.java

Here are Metacard examples.
https://github.com/codice/ddf/blob/master/catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/MetacardActionRegistry.java
https://github.com/codice/ddf/blob/master/catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/AbstractMetacardActionProvider.java

Reply:
I put the actions in SourceDescriptor because it felt like a natural fit. The alternative would be to have the RESTEndpoint look-up the actions from the action registry.